src/GridFS/ReadableStream.php

```diff
@@ -47,7 +48,8 @@ class ReadableStream
 
     private int $chunkOffset = 0;
 
-    private ?Cursor $chunksIterator = null;
+    /** @var (CursorInterface&Iterator)|null */
+    private $chunksIterator = null;
```

Comment:
You can set one of the union type or does that make a psalm error?
```suggestion
    private Iterator $chunksIterator = null;
```

Reply:
Using `?Iterator` works as expected.